mne_icalabel/iclabel/tests/test_label_components.py

```diff
@@ -13,12 +15,47 @@
 raw.filter(l_freq=1.0, h_freq=100.0)
 raw.set_eeg_reference("average")
 # fit ICA
-ica = ICA(n_components=15, method="picard")
+ica = ICA(n_components=5, method="picard")
 ica.fit(raw)
 
 
 @pytest.mark.filterwarnings("ignore::RuntimeWarning")
 def test_label_components():
     """Simple test to check that label_components runs without raising."""
     labels = iclabel_label_components(raw, ica)
-    assert labels is not None
+    assert labels.shape == (ica.n_components_, 7)
+
+
+def test_warnings():
+    """Test warnings issued when the raw|epochs|ica instance are not using the
+    same algorithm/reference/filters as ICLabel."""
+    data = np.random.randint(low=1, high=10, size=(3, 5000))
+    raw = RawArray(data, create_info(["Fpz", "CPz", "Oz"], sfreq=500, ch_types="eeg"))
+    raw.set_montage("standard_1020")
+    raw.filter(1.0, None)
+
+    # wrong raw, correct ica
+    ica = ICA(n_components=3, method="infomax", fit_params=dict(extended=True))
+    ica.fit(raw)
+    with pytest.warns(RuntimeWarning, match="common average reference"), pytest.warns(
+        RuntimeWarning, match="not filtered between 1 and 100 Hz"
+    ):
+        iclabel_label_components(raw, ica)
```

Comment:
Looks like this part isn't firing a warning. Not sure why tho...

error msg from CI
```
with pytest.warns(RuntimeWarning, match="common average reference"), pytest.warns(
[69](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:69)
            RuntimeWarning, match="not filtered between 1 and 100 Hz"
[70](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:70)
        ):
[71](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:71)
>           iclabel_label_components(raw, ica)
[72](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:72)
E           Failed: DID NOT WARN. No warnings of type (<class 'RuntimeWarning'>,) were emitted. The list of emitted warnings is: [].
[73](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:73)

[74](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:74)
data       = array([[4, 9, 2, ..., 1, 2, 5],
[75](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:75)
       [6, 4, 6, ..., 1, 2, 7],
[76](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:76)
       [1, 9, 2, ..., 3, 9, 7]])
[77](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:77)
ica        = <ICA | raw data decomposition, method: infomax (fit in 500 iterations on 5000 samples), 3 ICA components explaining 100.0 % of variance (3 PCA components available), channel types: eeg, no sources marked for exclusion>
[78](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:78)
raw        = <RawArray | 3 x 5000 (10.0 s), ~128 kB, data loaded>
[79](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:79)

[80](https://github.com/mne-tools/mne-icalabel/runs/6342837668?check_suite_focus=true#step:9:80)
mne_icalabel/iclabel/tests/test_label_components.py:43: Failed
```

Reply:
It is firing, it's just not getting caught by pytest for some reason.. not sure why either..

Reply:
Is it working locally? / are the other warnings firing?

Reply:
@larsoner any chance you know where this might be coming from? My understanding is that `mne.utils.warn` should fire by default.

Reply:
I pushed a fix, but I don't know why the syntax:

```
with pytest.warns(RuntimeWarning, match=...), pytest.warns(RuntimeWarning, match=...):
    function()
```

does not work.

Reply:
I'm still interesting to figure out why this syntax was failing, if anyone as any idea...
I wonder if parenthesis were necessary, e.g.

````
with (pytest.warns(RuntimeWarning, match=...), pytest.warns(RuntimeWarning, match=...)):
    function()
```

Reply:
I have not looked at the error, but the syntax seems fine, but I wouldn't expect it to work since each instance will try to caputure/swallow all warnings without allowing them to propagate (I think), but only one of them will be able to do so (the other one will get no warnings because they've been "taken" by the other one)